### PR TITLE
Improve Tests for Implicit RIDs and Libraries

### PR DESCRIPTION
**Pull request by nagilson**
As a follow-up to https://github.com/dotnet/sdk/pull/28628, this PR responds to PR feedback for tests that we didn't manage to get ready in time, as it would have delayed builds too long and we needed to get the fix in ASAP. Confirmed that the new tests  fail in both regards without the change and pass in both regards (library paths and RID properties are set correctly) in the new scenario. We could target it to 7.0.1xx but I think that's locked down atm. 

Drafting until 7.0.1xx changes have merged into 7.0.2xx. 
cc @dsplaisted

### CODE REVIEW

**Comment by dsplaisted** [comment]
I think we can probably target this to 7.0.1xx.  Test changes don't have to be approved.  We would probably just have to wait until the branch is open in to merge the PR, to avoid triggering unwanted builds.

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs:65]: I'm not sure you need both this file existence check and the `RuntimteIdentifier` property value check.  I think that's mostly redundant.

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs:67]: I'd suggest setting `AppendRuntimeIdentifierToOutputPath` to false in the referenced project.  Otherwise you have to pass the RuntimeIdentifier to the `GetPropertyValues` method so that it will look in the right output path for the file that's written with the values.

File: src/Tests/Microsoft.NET.Publish.Tests/RuntimeIdentifiersTests.cs
Comment (dsplaisted) [inline src/Tests/Microsoft.NET.Publish.Tests/RuntimeIdentifiersTests.cs:197]: Typo:

```suggestion
        public void ImplicitRuntimeIdentifierOptOutCorrectlyOptsOut()
```

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs
Comment (nagilson) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs:20]: That's correct!

But I was considering adding the publish and build command to the TestProject class:
Something like `public CommandResult TestProject::Build(List<string> args,  bool shouldPass=true, (args would be optional through an overload without args))` and  `public CommandResult TestProject::Publish(List<string> args, bool shouldPass=true)`, and probably they would call a more generic function like `internal CommandResult TestProject::DotnetCommand(command, bool shouldPass=true)`? 

What do you think about doing that? I can see myself writing the same code a lot (and a lot of test code currently does this).

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs
Comment (nagilson) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs:65]: Yeah, might as well get rid of it (ha, funny pun) to make the path changes easier

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs
Comment (nagilson) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs:67]: I don't think we should do this, for `_testProject` it would need the RID but for `_referenceProject` the folder without the RID contains `PropertyValuesFile.txt` (or w/e it was called) even without `AppendruntimeIdentifierToOutputPath`. And if `AppendRuntimeIdentifierToOutputPath` was false then it would be a less comprehensive test, as that was one part of the problem that caused the failure in VS

Comment (nagilson) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs:67]: We discussed this and the reason is that if the test fails it will be harder to tell why it failed with `AppendRuntimeIdentifierToOutputPath` so I should set it to false.

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs
Comment (nagilson) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs:20]: We discussed this and I'm going to try to do it but it might be a problem because of the `_testAssetsManager`.

Comment (nagilson) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs:20]: it's not worth doing the above proposal, just going to merge into one function

File: src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs
Comment (nagilson) [reply src/Tests/Microsoft.NET.Publish.Tests/GivenThatWeWantToPublishASingleFileLibrary.cs]: We dont need to worry about this when we move rids out of propertyvalues.txt path.

**Comment by nagilson** [comment]
Will pass once https://github.com/dotnet/sdk/pull/28717 is merged. cc @dsplaisted  to approve. Confirmed the test failed before and passes with the newer rid inference code.

**Review by dsplaisted — APPROVED** [review]

**Pull request merged**
